src/runtime/scheduler.go

```diff
@@ -250,8 +258,13 @@ func scheduler() {
 				// It would be nice if we could detect deadlocks here, because
 				// there might still be functions waiting on each other in a
 				// deadlock.
+				// Due to interrupts, we may actually not be done - so we need to sleep.
 				scheduleLog("  no tasks left!")
-				return
+				sleepTicks(0)
```

Comment:
Why this sleep?

Reply:
The goal here was to wait until an event of some form (e.g. interrupt) comes in. In the future, this chunk of code will be replaced by `wait()`, but that doesn't exist yet. I really do not know if this is the right means of waiting - I don't know that we actually have the right mechanism for waiting. Maybe this should be some arbitrary amount of time?

Reply:
Ah yes, now I see. Instead of sleeping, I think this is the place where we should in fact wait until the next interrupt, which implies the `wfi` or `wfe` instruction on ARM, `wfi` on RISC-V, and configuring the `SMCR` register and issuing `sleep` on AVR.

Reply:
Is there any particular solution you want me to use for now?

Reply:
Is there any particular solution you want me to use for now?

Reply:
This will be fixed by #680, but this needs to be merged first.

Reply:
Ah, now I understand the coupling you mentioned.